flink-table/flink-table-common/src/main/java/org/apache/flink/table/data/StringData.java

```diff
@@ -23,43 +23,48 @@
 import org.apache.flink.table.types.logical.VarCharType;
 
 /**
- * {@link StringData} is an internal data structure represents data of {@link VarCharType}
- * and {@link CharType} in Flink Table/SQL.
+ * An internal data structure representing data of {@link CharType} and {@link VarCharType}.
+ *
+ * <p>This data structure is immutable.
```

Comment:
BinaryStringData is mutable, because it is lazy (de)serialized. The binary is lazily materialized when needed, so it's not thread-safe.

Reply:
@wuchong What I meant with immutable is that it does not provide any setters. Because `GenericRowData` has setters. But I'm fine with removing it here.